apis/r/src/Makevars.in

```diff
@@ -5,6 +5,7 @@ PKG_CPPFLAGS = -I. -I../inst/include/ @tiledb_include@ @cxx17_macos@ -Iinclude
 
 ## When this becomes a CRAN package we may have to remove this. For now it keeps the noise down
 PKG_CXX17FLAGS = -Wno-deprecated-declarations
+PKG_CLANGFLAGS = -Wno-deprecated-declarations
```

Comment:
That option is wrong and does not exist.

R uses `PKG_CXX*FLAGS` and allows different ones for different compilation standards.  And C++17 is used here per an earlier line in the same file.   

So I suggest to remove this.

Reply:
Thanks @eddelbuettel -- this PR is still in draft with no reviewers -- when I build (using `clang`) the `-Wno-deprecated-declarations` doesn't actually get found ... this was worth a try ...

Reply:
Ah yes I feared that.  It's an option that CRAN disallows in packages for that reason.  Dang.  I guess we could move a test to `configure` and act based on 

     compiler=`${R_HOME}/bin/R CMD config CXX17`